app/shop.py

```diff
@@ -0,0 +1,38 @@
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int],
+            products: dict[str, int | float]
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def make_purchases(self, customer: object) -> None:
```

Comment:
```suggestion
    def make_purchases(self, customer: Customer) -> None:
```
`object` is too broad annotation